torch/testing/_internal/common_device_type.py

```diff
@@ -1351,8 +1351,23 @@ def largeTensorTest(size, device=None):
     def inner(fn):
         @wraps(fn)
         def dep_fn(self, *args, **kwargs):
-            size_bytes = size(self, *args, **kwargs) if callable(size) else size
-            _device = device if device is not None else self.get_primary_device()
+            size_bytes: int = size(self, *args, **kwargs) if callable(size) else size
+            _device = device
+            if _device is None:
+                if hasattr(self, "get_primary_device"):
+                    _device = self.get_primary_device()
+                else:
+                    _device = self.device
+
+            # If this is running with GPU cpp_wrapper, the autotuning step will generate
+            # an additional array of the same size as the input.
+            if (
+                TEST_WITH_TORCHINDUCTOR
+                and torch._inductor.config.cpp_wrapper
+                and _device != "cpu"
+            ):
+                size_bytes *= 2
```

Comment:
Note to reviewers: multiplying by 2 is conservative; many use cases will be less than this. This represents the worst-case likely scenario.